test/test_helper.rb

```diff
@@ -62,7 +62,9 @@ def log_hook(obj)
     def hook_stdin(io)
       org_stdin = $stdin
       $stdin = io
-      yield.tap { $stdin = org_stdin }
+      yield
```

Comment:
nit: I would say it's more clear to wrap the yield statement only, instead of the whole function. For example, what if the statement `org_stdin = $stdin` failed? We should not try to restore `$stdin` in that case.